firebase-ai/src/main/kotlin/com/google/firebase/ai/type/ImagenMaskReference.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.firebase.ai.type
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Rect
+
+@PublicPreviewAPI
+public abstract class ImagenMaskReference
+internal constructor(maskConfig: ImagenMaskConfig, image: ImagenInlineImage? = null) :
+  ImagenReferenceImage(maskConfig = maskConfig, image = image) {
+
+  public companion object {
+    public fun generateMaskAndPadForOutpainting(
+      image: ImagenInlineImage,
+      newDimensions: Dimensions,
+      newPosition: ImagenImagePlacement = ImagenImagePlacement.CENTER,
+    ): List<ImagenReferenceImage> {
+      val originalBitmap = image.asBitmap()
+      val normalizedPosition =
+        newPosition.normalizeToDimensions(
+          Dimensions(originalBitmap.width, originalBitmap.height),
+          newDimensions,
+        )
+      val normalizedImageRectangle =
+        Rect(
+          normalizedPosition.x!!,
```

Comment:
could you handle the case of `x` or `y` being `null` and then use them without the `!!`? It's preferred not to use it.

Reply:
done, although after normalization it is impossible that they actually be null.